dev/Repeater/APITests/FlowLayoutCollectionChangeTests.cs

```diff
@@ -306,6 +306,78 @@ public void CanReplaceSingleItem()
             });
         }
 
+        [TestMethod]
+        public void CanMoveItem()
+        {
+            CustomItemsSource dataSource = null;
+            RunOnUIThread.Execute(() => dataSource = new CustomItemsSource(Enumerable.Range(0, 10).ToList()));
+            ScrollViewer scrollViewer = null;
+            ItemsRepeater repeater = null;
+            var viewChangedEvent = new ManualResetEvent(false);
+            int elementsCleared = 0;
+            int elementsPrepared = 0;
+            int elementsIndexChanged = 0;
+
+            RunOnUIThread.Execute(() =>
+            {
+                repeater = SetupRepeater(dataSource, ref scrollViewer);
+                scrollViewer.ViewChanged += (sender, args) =>
+                {
+                    if (!args.IsIntermediate)
+                    {
+                        viewChangedEvent.Set();
+                    }
+                };
+
+                repeater.ElementPrepared += (sender, args) => { elementsPrepared++; };
+                repeater.ElementIndexChanged += (sender, args) => { elementsIndexChanged++; };
+                repeater.ElementClearing += (sender, args) => { elementsCleared++; };
+
+                scrollViewer.ChangeView(null, 300, null, true);
```

Comment:
Is the ChangeView relevant for this fix ? If you can avoid the ChangeView, you can put the entire test within a single RunOnUIThread block and simplify a lot of this code.

Reply:
I think we need the changeview to be able to test moves between the realized range, here I was trying to follow the example of the CanReplaceSingleItem test. But if you think it is not necessary to be that through here I could remove the ChangeView + "the move before realized range" test.

Reply:
ok got it. I think it is testing combinations of inside and outside the realized range.